engine/swarm/swarm_manager_locking.md

```diff
@@ -0,0 +1,162 @@
+```
+
+Record the key in a safe place.
+
+When Docker restarts, a new manager node attempts to join the swarm, or Docker
+needs to load the encryption key from disk to a node's memory, you need to
+[unlock the swarm](swarm_manager_locking.md#unlock-a-swarm). You will see an
+error like the following and services will not start.
+
+```bash
+$ sudo service docker restart
+
+$ docker service ls
+
+Error response from daemon: Swarm is encrypted and needs to be unlocked before it can be used. Please use "docker swarm unlock" to unlock it.
+```
+
+## Enable or disable autolock on an existing swarm
+
+To enable autolock on an existing swarm, set the `autolock` flag to `true`.
+
+```bash
+$ docker swarm update --autolock="true"
+
+Swarm updated.
+To unlock a swarm manager after it restarts, run the `docker swarm unlock`
+command and provide the following key:
+
+    SWMKEY-1-+MrE8NgAyKj5r3NcR4FiQMdgu+7W72urH0EZeSmP/0Y
+
+Please remember to store this key in a password manager, since without it you
+will not be able to restart the manager.
+```
+
+To disable autolock, set `--autolock` to `false`. The mutual TLS key and the
+encryption key used to read and write RAFT logs will be stored unencrypted on
+disk. There is a trade-off between the risk of storing the encryption key
+unencrypted at rest and the convenience of being able to restart a swarm without
+needing to unlock the swarm.
+
+```bash
+$ docker swarm update --autolock="false"
+```
+
+Keep the unlock key around for a short time after disabling autolocking, in case
+a manager goes down while it is still configured to lock using the old key.
+
+## Unlock a swarm
+
+To unlock a locked swarm, use `docker swarm unlock`.
+
+```bash
+$ docker swarm unlock
+
+Please enter unlock key:
+```
+
+Enter the encryption key that was generated and shown in the command output when
+you locked the swarm or rotated the key, and the swarm unlocks.
+
+## View the current unlock key for a running swarm
+
+Consider a situation where your swarm is
+running as expected, then manager node becomes unavailable. You troubleshoot the
+problem and bring the physical node back online, but you need to rejoin the
+node to the swarm. In order to do this, you need to provide the unlock key, so
+that the node can load the encrypted unlock key into memory and decrypt it.
+
+If the key has not been rotated since the node left the swarm, and you have a
+quorum of functional manager nodes in the swarm, you can view the current unlock
+key using `docker swarm unlock-key` without any arguments.
+
+```bash
+$ docker swarm unlock-key
+
+To unlock a swarm manager after it restarts, run the `docker swarm unlock`
+command and provide the following key:
+
+    SWMKEY-1-8jDgbUNlJtUe5P/lcr9IXGVxqZpZUXPzd+qzcGp4ZYA
+
+Please remember to store this key in a password manager, since without it you
+will not be able to restart the manager.
+```
+
+If you ever need to view the unlock key in this way,
```

Comment:
Doesn't make a lot of sense, since rotating a swarm also displays the new key. I think we should remove it, or at least replace it with a warning around: please don't share this key anywhere, and store it securely, etc`